### PR TITLE
Q4 perchannel

### PR DESCRIPTION
**Pull request by luoyu-intel**
## Type of Change

add q4 per channel quant and kernel.
fix q4 quant low accuracy bug.

## How has this PR been tested?

llama-7b gptj-6b

### CODE REVIEW

**Review by airMeng — APPROVED** [review]

File: intel_extension_for_transformers/llm/runtime/graph/core/layers/ip_fusion_ffn.cpp [hunk 1]
```diff
@@ -496,7 +554,90 @@ JBLAS_CODE jblas_fusion_FFN_Add_GeLu_s8fp32pern_f32f32_forward(float* activation
...
+      int ldo = fout;
+      // FusedInter::Arguments::paramA paramA={activation, lda};
+      // FusedInter::Arguments::paramW1 paramW1={w1tmp};
+      // FusedInter::Arguments::paramW2 paramW2={w2tmp};
+      // FusedInter::Arguments::param1 param1={tmp1, b1ptr, ldtmp1, ldtmp1};
+      auto quanA1 = finter.getActivationPtr()->createStorage(seq, fin, (int8_t*)workspace);
+      auto offset = workspace == NULL ? 0 : finter.getActivationPtr()->getWorkSpaceSize(seq, fin);
+      auto quanA2 = finter.getActivationPtr()->createStorage(seq, fmid, (int8_t*)workspace + offset);
+      ret = finter.compute({seq,
+                            fin,
```
Comment (airMeng) [inline intel_extension_for_transformers/llm/runtime/graph/core/layers/ip_fusion_ffn.cpp:611]: should we unify format same as line570?

Comment (airMeng) [reply intel_extension_for_transformers/llm/runtime/graph/core/layers/ip_fusion_ffn.cpp:611]: quite wield, are you using clang-format 14.0.0?

File: intel_extension_for_transformers/llm/library/jblas/jblas/jit_blas_utils.h [hunk 2]
```diff
@@ -37,6 +37,7 @@
 #define CompileAVX512F() _MSC_VER && (_MSC_VER >= 1911)
 #define CompileAVX2() _MSC_VER && (_MSC_VER >= 1900)
 #define CompileAMX() 0
+#define CompileBF16Kernel() 0
```
Comment (DDEle) [inline intel_extension_for_transformers/llm/library/jblas/jblas/jit_blas_utils.h:40]: Added it back? But only for windows?

Comment (luoyu-intel) [reply intel_extension_for_transformers/llm/library/jblas/jblas/jit_blas_utils.h]: removed

**Comment by luoyu-intel** [comment]
gptj-6b s4 perchannel 32(prompt) perf: 19ms/token
 
model_print_timings:      sample time =    12.46 ms /    16 runs   (    0.78 ms per token)
model_print_timings: prompt eval time =    70.07 ms /    32 tokens (    2.19 ms per token)
model_print_timings:        eval time =   283.87 ms /    15 runs   (   18.92 ms per token)
model_print_timings:       total time =  1368.08 ms
========== eval time log of each prediction ==========
prediction   0, time: 70.07ms
prediction   1, time: 19.40ms
prediction   2, time: 18.99ms
prediction   3, time: 18.98ms
prediction   4, time: 18.96ms
prediction   5, time: 18.82ms
prediction   6, time: 18.88ms

**Comment by luoyu-intel** [comment]
gptj-6b s4 perchannel (prompt len=1919) perf: 25ms/token

model_print_timings: prompt eval time =  1035.76 ms /  1919 tokens (    0.54 ms per token)
model_print_timings:        eval time =   374.96 ms /    15 runs   (   25.00 ms per token)
model_print_timings:       total time =  2383.93 ms
========== eval time log of each prediction ==========
prediction   0, time: 1035.76ms
prediction   1, time: 26.38ms
prediction   2, time: 25.62ms
prediction   3, time: 25.38ms
prediction   4, time: 25.25ms
prediction   5, time: 25.25ms
prediction   6, time: 25.42ms

**Review by DDEle — APPROVED** [review]
LGTM

**Comment by luoyu-intel** [comment]
llama-7b q4 per channel(prompt len=32) perf:

model_print_timings: prompt eval time =    71.21 ms /    34 tokens (    2.09 ms per token)
model_print_timings:        eval time =   317.68 ms /    15 runs   (   21.18 ms per token)
model_print_timings:       total time =  1349.90 ms
========== eval time log of each prediction ==========
prediction   0, time: 71.21ms
prediction   1, time: 21.76ms
prediction   2, time: 21.43ms
prediction   3, time: 21.21ms
prediction   4, time: 21.19ms
prediction   5, time: 21.15ms
prediction   6, time: 21.28ms
prediction   7, time: 21.06ms
prediction   8, time: 21.01ms

**Comment by luoyu-intel** [comment]
llama-7b q4 per channel(prompt len=1790) perf: 24.3ms/token

model_print_timings: prompt eval time =   927.81 ms /  1790 tokens (    0.52 ms per token)
model_print_timings:        eval time =   365.15 ms /    15 runs   (   24.34 ms per token)
model_print_timings:       total time =  2342.39 ms
========== eval time log of each prediction ==========
prediction   0, time: 927.81ms
prediction   1, time: 24.84ms
prediction   2, time: 24.43ms
prediction   3, time: 24.22ms
prediction   4, time: 25.16ms
prediction   5, time: 24.24ms
prediction   6, time: 24.36ms
prediction   7, time: 24.32ms

**Pull request merged**